Mathlib/Algebra/Polynomial/Basic.lean

```diff
@@ -1168,6 +1168,10 @@ instance [IsCancelAdd R] [IsCancelMulZero R] : IsCancelMulZero R[X] where
 
 instance [IsCancelAdd R] [IsDomain R] : IsDomain R[X] where
 
+theorem noZeroDivisors_iff : NoZeroDivisors R[X] ↔ NoZeroDivisors R where
```

Comment:
Can you add a doc giving a link to `isDomain_iff` and to `notMem_nonZeroDivisors_iff`. Also with a short explanation why those are no contradictory (this took me a little to figure out since I am used to "being a domain" = "not having zero divisors"but this of course fails over a semiring.

Reply:
Sorry, I missed your comment. I just mean that mathematicians are not used to work with semirings so the statement seems weird since (as you did) it allows you to build a nondomain without zero divisors. I am just saying that most of the people will say "wait, what?!"